lib/Newsletter/Renderer/Blocks/Text.php

```diff
@@ -120,8 +120,8 @@ public function convertParagraphsToTables($html) {
         $lineBreaks;
       if ($paragraph->hasClass(PostContentManager::WP_POST_CLASS)) {
         $paragraph->removeClass(PostContentManager::WP_POST_CLASS);
-        // if this element is followed by a paragraph, add double line breaks
-        $lineBreaks = ($nextElement && preg_match('/<p/i', $nextElement->getOuterText())) ?
+        // if this element is followed by a paragraph or heading, add double line breaks
+        $lineBreaks = ($nextElement && preg_match('/<(p|h[1-6]{1})/i', $nextElement->getOuterText())) ?
```

Comment:
Maybe this behavior could be tested within [the text block](https://github.com/mailpoet/mailpoet/blob/deb4d7833d0140a7a9381082146ec228d05897ca/tests/unit/Newsletter/Renderer/Blocks/TextTest.php) unit test.